File: lib/ui/scenic/tests/scenic_test.cc
// Copyright 2018 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "garnet/lib/ui/scenic/tests/scenic_test.h"

namespace scenic {
namespace test {

std::unique_ptr<component::StartupContext> ScenicTest::app_context_;

void ScenicTest::SetUp() {
  // TODO(SCN-720): Wrap CreateFromStartupInfo using ::gtest::Environment
  // instead of this hack.  This code has the chance to break non-ScenicTests.
  if (app_context_ == nullptr) {
    app_context_ = component::StartupContext::CreateFromStartupInfo();
  }
  scenic_ =
      std::make_unique<Scenic>(app_context_.get(), [this] { QuitLoop(); });
  InitializeScenic(scenic_.get());
}

void ScenicTest::TearDown() {
  reported_errors_.clear();
  events_.clear();
  scenic_.reset();
}

void ScenicTest::InitializeScenic(Scenic* scenic) {}

void ScenicTest::ReportError(fxl::LogSeverity severity,
                             std::string error_string) {
// Typically, we don't want to log expected errors when running the tests.
// However, it is useful to print these errors while writing the tests.
#if 0
  switch (severity) {
    case ::fxl::LOG_INFO:
      FXL_LOG(INFO) << error_string;
      break;
    case ::fxl::LOG_WARNING:
      FXL_LOG(WARNING) << error_string;
      break;
    case ::fxl::LOG_ERROR:
      FXL_LOG(ERROR) << error_string;
      break;
    case ::fxl::LOG_FATAL:
      FXL_LOG(FATAL) << error_string;
      break;
  }
#endif
  reported_errors_.push_back(error_string);
}

void ScenicTest::EnqueueEvent(fuchsia::ui::scenic::Event event) {
  events_.push_back(std::move(event));
}

}  // namespace test
}  // namespace scenic

File: src/tests/end_to_end/sl4f/test/scenic_test.dart
// Copyright 2019 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:test/test.dart';
import 'package:sl4f/sl4f.dart' as sl4f;

const _timeout = Timeout(Duration(minutes: 1));

void main() {
  sl4f.Sl4f sl4fDriver;
  sl4f.Scenic scenic;

  setUp(() async {
    sl4fDriver = sl4f.Sl4f.fromEnvironment();
    await sl4fDriver.startServer();

    scenic = sl4f.Scenic(sl4fDriver);
  });

  tearDown(() async {
    await sl4fDriver.stopServer();
    sl4fDriver.close();
  });

  group(sl4f.Scenic, () {
    test('talks to SL4F server without error', () {
      expect(scenic.takeScreenshot(), completion(isNotNull));
    });
  }, timeout: _timeout);
}
